src/qt/pivx/settings/settingsexportcsv.cpp

```diff
@@ -55,32 +51,38 @@ SettingsExportCSV::SettingsExportCSV(PIVXGUI* _window, QWidget *parent) :
     setFilterAddressBook(ui->comboBoxSortAddressType, lineEditAddressBook);
     ui->comboBoxSortAddressType->setCurrentIndex(0);
 
-    connect(ui->pushButtonSave, &QPushButton::clicked, this, &SettingsExportCSV::exportClicked);
     connect(ui->pushButtonDocuments, &QPushButton::clicked, [this](){selectFileOutput(true);});
+    connect(this, &SettingsExportCSV::fileSelectedTxes, this, &SettingsExportCSV::exportTxes);
 
-    connect(ui->pushButtonExportAddress, &QPushButton::clicked, this, &SettingsExportCSV::onExportAddressesClicked);
     connect(ui->pushButtonAddressDocuments, &QPushButton::clicked, [this](){selectFileOutput(false);});
+    connect(this, &SettingsExportCSV::fileSelectedAddrs, this, &SettingsExportCSV::exportAddrs);
 }
 
 void SettingsExportCSV::selectFileOutput(const bool& isTxExport)
 {
-    QString filenameRet = GUIUtil::getSaveFileName(this,
+    QString filename = GUIUtil::getSaveFileName(this,
                                         isTxExport ? tr("Export CSV") : tr("Export Address List"), QString(),
                                         isTxExport ? tr("PIVX_tx_csv_export(*.csv)") : tr("PIVX_addresses_csv_export(*.csv)"),
                                         nullptr);
 
-    if (!filenameRet.isEmpty()) {
-        if (isTxExport) {
-            filename = filenameRet;
+    if (isTxExport) {
+        if (!filename.isEmpty()) {
             ui->pushButtonDocuments->setText(filename);
+            Q_EMIT fileSelectedTxes(filename);
```

Comment:
Remember that you are in the same object running in the same thread, no need to create a signal/slot connection. The method can be called directly.

Reply:
good point